### PR TITLE
Feature: preferences

### PR DESCRIPTION
**Pull request by JonathanBottcher**
#### Board:
* [Ticket #NUMBER_OF_THE_TICKET](link_goes_here)
---
#### Description:

---
#### Notes:
*
---
#### Tasks:
- [x] Add each element in this format
---
#### Risk:
*
---

### CODE REVIEW

File: app/controllers/preferences_controller.rb [hunk 1]
```diff
@@ -1,8 +1,7 @@
-# frozen_string_literal: true
-
 class PreferencesController < ApplicationController
   def index
+    binding.pry
```
Comment (enzofab91) [inline app/controllers/preferences_controller.rb]: Please remove this

File: app/controllers/preferences_controller.rb [hunk 2]
```diff
@@ -1,8 +1,7 @@
...
     @preferences = current_user.preferences
     @pagy, @records = pagy(@preferences)
   end
-end
+end
```
Comment (enzofab91) [inline app/controllers/preferences_controller.rb]: Add a new line here

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -57,4 +57,5 @@ def user_params
   def set_user
     @user = User.find(params[:id])
   end
+
```
Comment (enzofab91) [inline app/controllers/users_controller.rb]: 👀

File: app/models/preference.rb [hunk 4]
```diff
@@ -1,2 +1,21 @@
...
+#  restriction :boolean          default(TRUE), not null
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+# Indexes
+#
+#  index_preferences_on_user_id  (user_id)
+#
 class Preference < ApplicationRecord
+  MAX_PREFERENCES = 10
```
Comment (enzofab91) [inline app/models/preference.rb]: This is not in the scope of this ticket so you can leave it like this and tackle later, but I believe the limit is 5 😅

File: db/migrate/20241021175734_create_preferences.rb [hunk 5]
```diff
@@ -1,9 +1,10 @@
 class CreatePreferences < ActiveRecord::Migration[7.1]
   def change
     create_table :preferences do |t|
+      t.references :user, foreign_key: true, null: false
       t.string :name
```
Comment (enzofab91) [inline db/migrate/20241021175734_create_preferences.rb]: If you are validating `name` presence in the model, here would be nice to not allow null in `name` as well. wdyt?

File: spec/models/preference_spec.rb [hunk 6]
```diff
@@ -1,3 +1,19 @@
...
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+# Indexes
+#
+#  index_preferences_on_user_id  (user_id)
+#
 require 'rails_helper'
 
 RSpec.describe Preference, type: :model do
```
Comment (enzofab91) [inline spec/models/preference_spec.rb:19]: What about adding some test to check validations? Check [be_valid](https://www.rubydoc.info/gems/rspec-rails/RSpec%2FRails%2FMatchers:be_valid) and [validate_presence_of](https://github.com/thoughtbot/shoulda-matchers/blob/main/lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb) helpers.

Also user model has already tests implemented 😉

**Review by enzofab91 — COMMENTED** [review]
Nice work!! Left some comments.

Also what if you update PR title?

File: spec/models/preference_spec.rb
Comment (JonathanBottcher) [inline spec/models/preference_spec.rb:1]: I reckon tests should be implemented similar to the user test validations, but as for now I'll leave it as is until I fully understand that approach.

**Review by enzofab91 — APPROVED** [review]
Nice! 👏

**Pull request merged**